### PR TITLE
Backport PR #10729 on branch 3.2.x (Toc: Run nested code cells directly from markdown headings)

### PR DESCRIPTION
**Pull request by jess-x**
… headings

## References
https://github.com/jupyterlab/jupyterlab/pull/10729#issuecomment-944992145
Backport PR #10729 on branch 3.2.x (Toc: Run nested code cells directly from markdown headings)

### CODE REVIEW

File: packages/toc/src/utils/headings.ts [hunk 1]
```diff
@@ -94,59 +93,10 @@ const isNotebookHeading = (heading: any): heading is INotebookHeading => {
...
-  }
-};
-
 /**
  * Exports.
  */
 export { IHeading };
 export { INumberedHeading };
 export { INotebookHeading };
-export { runNestedCodeCells };
```
Comment (krassowski) [inline packages/toc/src/utils/headings.ts:152]: Should we keep this function (and its export) in this backport for backward compatibility?

Comment (jess-x) [reply packages/toc/src/utils/headings.ts:152]: Good point!! let me see how to add those back

Comment (jess-x) [reply packages/toc/src/utils/headings.ts:152]: Added them back manually :)

Comment (krassowski) [reply packages/toc/src/utils/headings.ts:152]: Thanks!

**Review by krassowski — APPROVED** [review]
Thank you! I will highlight the added `@jupyterlab/cells` dependency for anyone reviewing after me before merging (I assume it is ok). I don't believe the failing visual regression test is related, but I also do not understand why it is failing. Seems to be related to https://github.com/jupyterlab/jupyterlab/pull/11360 in a way...

**Comment by jess-x** [comment]
@krassowski thanks for the review!! 
Agree, the visual regression test doesn't look related to me either. JS flaky tests are also failing though  - I don't know it that is related, but remains me that I have noticed, even when my kernel is idle, `Run Cells` sometimes works intermittently. I might investigate that a bit separately, if more folk are also seeing that behavior.

**Comment by jtpio** [comment]
There were a couple of UI tests fixes in https://github.com/jupyterlab/jupyterlab/pull/11112.

Which is being backported to 3.2.x in https://github.com/jupyterlab/jupyterlab/pull/11371. So maybe landing that backport PR first will help.

**Comment by jtpio** [comment]
Just rebased to grab the latest fixes on the `3.2.x` branch.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by jess-x** [comment]
Thanks @jtpio!!

**Pull request merged**